src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java

```diff
@@ -244,51 +244,74 @@ private void parseSiegeWarSiegeImmunityCommand(CommandSender sender, String[] ar
 			showSiegeImmunityHelp(sender);
 			return;
 		}
-
 		try {
-			if (args[0].equalsIgnoreCase("alltowns"))
+			if (args[0].equalsIgnoreCase("alltowns") && !args[1].equalsIgnoreCase("permanent")) {
 				Integer.parseInt(args[1]);
-			else
+			} else if (!args[0].equalsIgnoreCase("alltowns") && !args[2].equalsIgnoreCase("permanent")){
 				Integer.parseInt(args[2]);
+			}
 		} catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
 			Messaging.sendMsg(sender, Translation.of("msg_error_must_be_num"));
 			showSiegeImmunityHelp(sender);
 			return;
 		}
 
+		String timeDuration;
 		if (args.length >= 3 && args[0].equalsIgnoreCase("town")) {
 			//town {townname} {hours}
 			Town town = TownyUniverse.getInstance().getTown(args[1]);
 			if (town == null) {
 				Messaging.sendErrorMsg(sender, Translation.of("msg_err_not_registered_1", args[1]));
 				return;
 			}
-			long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
-			TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
-			TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-
-		} else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
+			if (args[2].equalsIgnoreCase("permanent")) {
+				TownMetaDataController.setSiegeImmunityEndTime(town, -1l);
+				timeDuration = "permanent";
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				timeDuration = (Long.parseLong(args[2]) == 1L) ? "1 hour" : String.format("%s hours", args[2]);

```

Comment:
:construction: 
- This formula needs a fix, because it has hardcoded english strings 
- Also it doesn't need to account for single hours (*but you can support this is you wish*)
- I believe something like this would work: 
>   timeDuration = Long.parseLong(args[1]) + " " + Towny.Translation.of("Hours").